module/zfs/zfs_ctldir.c

```diff
@@ -1064,7 +1054,8 @@ zfsctl_snapshot_mount(struct path *path, int flags)
 	zfs_sb_t *snap_zsb;
 	zfs_snapentry_t *se;
 	char *full_name, *full_path;
-	char *argv[] = { "/bin/sh", "-c", NULL, NULL };
+	char *argv[] = { "/usr/libexec/zfs/z_usermodehelper", "mount",
```

Comment:
One of the problem with adding a `z_usermodehelper` command is that we'd need to install it in a standard location on all systems.  That would need to be either `/bin/` or `/sbin/`, a directory like `/usr/libexec` for example won't exist on any Debian derived distributions.

But let me again ask the question why can't we directly call `/sbin/mount.zfs` here?  Similarly, there's nothing preventing us from adding a `/sbin/umount.zfs` helper and using that.  I want to avoid adding a custom helper which is built specifically for this purpose.